Mathlib/Data/List/MinMax.lean

```diff
@@ -392,25 +392,66 @@ theorem coe_le_maximum_iff : a ≤ l.maximum ↔ ∃ b, b ∈ l ∧ a ≤ b := b
   induction l with
   | nil => simp
   | cons h t ih =>
-    simp [List.maximum_cons, ih]
+    simp [maximum_cons, ih]
 
-theorem minimum_le_coe_iff : l.minimum ≤ a ↔ ∃ b, b ∈ l ∧ b ≤ a := by
-  induction l with
-  | nil => simp
-  | cons h t ih =>
-    simp [List.minimum_cons, ih]
+theorem minimum_le_coe_iff : l.minimum ≤ a ↔ ∃ b, b ∈ l ∧ b ≤ a :=
+  @coe_le_maximum_iff αᵒᵈ _ _ _
 
 theorem maximum_ne_bot_of_ne_nil {l : List α} (h : l ≠ []) : l.maximum ≠ ⊥ :=
-  match l, h with | _ :: _, _ => by simp [List.maximum_cons]
+  match l, h with | _ :: _, _ => by simp [maximum_cons]
 
 theorem minimum_ne_top_of_ne_nil {l : List α} (h : l ≠ []) : l.minimum ≠ ⊤ :=
-  @List.maximum_ne_bot_of_ne_nil αᵒᵈ _ _ h
+  @maximum_ne_bot_of_ne_nil αᵒᵈ _ _ h
 
 theorem maximum_ne_bot_of_length_pos {l : List α} (h : 0 < l.length) : l.maximum ≠ ⊥ :=
-  match l, h with | _ :: _, _ => by simp [List.maximum_cons]
+  match l, h with | _ :: _, _ => by simp [maximum_cons]
 
 theorem minimum_ne_top_of_length_pos {l : List α} (h : 0 < l.length) : l.minimum ≠ ⊤ :=
-  @List.maximum_ne_bot_of_length_pos αᵒᵈ _ _ h
+  @maximum_ne_bot_of_length_pos αᵒᵈ _ _ h
+
+/-- The maximum value in a non-empty `List`. -/
+def maximum_of_length_pos {L : List α} (h : 0 < L.length) : α :=
+  WithBot.unbot L.maximum (maximum_ne_bot_of_length_pos h)
```

Comment:
We already have `{l : List α}` from `variable` on line 342 so we can drop a lot of `{L : List α}` and `{l : List α}` below:
```suggestion
def maximum_of_length_pos (h : 0 < l.length) : α :=
  WithBot.unbot l.maximum (maximum_ne_bot_of_length_pos h)
```